src/main.rs

```diff
@@ -11,25 +7,36 @@ use std::io::Write;
 use std::process;
 
 use gb_io::reader::SeqReader;
-use regex::Regex;
-
-// Translation table 11 from https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi.
-// Indexing into this array is as follows:
-// T = 0, C = 1, A = 2, G = 3
-// index = (16 * first_base) + (4 * second_base) + third_base
-// So... TTT = 0, TTC = 1, TTA = 2, ... , GGC = 61, GGA = 62, GGG = 63
-const GENETIC_CODE: &str = "FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG";
-// some codons can be used as Met in the start position
-const STARTS: &str = "---M------**--*----M------------MMMM---------------M------------";
-
-// ERR_BAD_NT is an error value for an invalid nucleotide
+use gb_io::{feature_kind, qualifier_key};
+use once_cell::sync::Lazy;
+
+#[derive(thiserror::Error, Debug)]
+enum Error {
+    #[error("Something went wrong in translation! {0:?}")]
+    BadNucleotide(Vec<usize>),
+
+    #[error("Invalid amino acid: {0}")]
+    InvalidAminoAcid(char),
+}
```

Comment:
This is an `Error` enum, defined with the help of the [`thiserror` crate](https://github.com/dtolnay/thiserror) (which generates some useful code in this `#[derive(thiserror::Error)]` macro). The code can be handwritten, but tends to be repetitive, and `thiserror` is a fairly minimal dependency. More about how to use it for [error handling](https://doc.rust-lang.org/stable/book/ch09-00-error-handling.html) when we call it later in the code.